### PR TITLE
Create CAP CLI migration script. Update boolean meta value to use ord…

### PR DESCRIPTION
**Pull request by willgladstone**
…er on posts.

### CODE REVIEW

File: php/class-fm-bylines-cli.php [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+class FM_Bylines_CLI extends WP_CLI_Command {
+
+	/**
+	 * Migrate Co-Authors Plus Guest Authors and linked accounts to FM Bylines
+	 *
+	 */
+	public function migrate_coauthors() {
+		global $coauthors_plus;
+
+		$cap_terms = get_terms( $coauthors_plus->coauthor_taxonomy, array( 'hide_empty' => false ) );
```
Comment (timatron) [inline php/class-fm-bylines-cli.php]: should fail if CAP isn't enabled

File: php/class-fm-bylines-cli.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+		foreach ( $cap_terms as $cap_term ) {
+			$args = array(
+				'post_type' => $coauthors_plus->guest_authors->post_type,
+				'name' => $cap_term->slug,
+				'post_status' => 'publish',
+				'numberposts' => 1,
+			);
+			$cap_post = get_posts( $args );
+			if ( ! empty( $cap_post[0]->ID ) ) {
+				$cap_meta = apply_filters( 'fm_bylines_cli_cap_meta', get_post_meta( $cap_post[0]->ID ), $cap_post[0], $cap_term );
```
Comment (timatron) [inline php/class-fm-bylines-cli.php]: how about a function, filter in CLI script seems overkill. no?

File: php/class-fm-bylines-cli.php [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+				if ( empty( $byline_post[0]->ID ) ) {
+					// Create a byline post
+					$byline_args = array(
+						'post_title' => $cap_meta['cap-display_name'][0],
+						'post_name' => $byline_slug,
+						'post_type' => FM_Bylines()->name,
+						'post_status' => 'publish',
+					);
+					$byline_id = wp_insert_post( $byline_args );
```
Comment (timatron) [inline php/class-fm-bylines-cli.php]: dont you need to check that it was successful?

File: php/class-fm-bylines-cli.php [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+				$objects = get_objects_in_term( $cap_term->term_id, $coauthors_plus->coauthor_taxonomy );
+				foreach ( $objects as $object_id ) {
+					if ( get_post_type( $object_id ) !== $coauthors_plus->guest_authors->post_type ) {
+						$current_bylines = empty( get_post_meta( $byline_id, 'fm_bylines_author', true ) ) ? array() : get_post_meta( $byline_id, 'fm_bylines_author', true );
+						$new_byline_entry = array(
+							'byline_id' => $byline_id,
+							'fm_byline_type' => $coauthors_plus->coauthor_taxonomy,
+						);
+						$current_bylines[] = $new_byline_entry;
+						update_post_meta( $byline_id, 'fm_bylines_' . $coauthors_plus->coauthor_taxonomy, $current_bylines );
```
Comment (timatron) [inline php/class-fm-bylines-cli.php]: should we define a mapping of cap author types to bylines author types?

File: php/class-fm-bylines-cli.php [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+						update_post_meta( $byline_id, 'fm_bylines_' . $coauthors_plus->coauthor_taxonomy . '_' . $byline_id, count( $current_bylines ) );
+					}
+
+					// Clean legacy data. Don't delete by default.
+					if ( apply_filters( 'fm_bylines_cli_clean_legacy_cap_data', false ) ) {
```
Comment (timatron) [inline php/class-fm-bylines-cli.php]: why filters? make it a cmd line option more better

File: php/class-fm-bylines-cli.php [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+						wp_delete_object_term_relationships( $object_id, $coauthors_plus->coauthor_taxonomy );
+					}
+				}
+
+				// Clean legacy data. Don't delete by default.
+				if ( apply_filters( 'fm_bylines_cli_clean_legacy_cap_data', false ) ) {
```
Comment (timatron) [inline php/class-fm-bylines-cli.php]: cmd line arg, no filter for this plz

**Comment by timatron** [comment]
a hoy matey, i be the toob fairy, but i need more argggggssss. 

:hocho:

File: php/class-fm-bylines-cli.php (hunk 1 above)
Comment (willgladstone) [inline php/class-fm-bylines-cli.php]: I'm not sure what you mean by fail if not enabled. This doesn't require to have cap enabled. Just that there is data. If you want me to stick in an error message if cap_terms is empty I can do that.

File: php/class-fm-bylines-cli.php (hunk 2 above)
Comment (willgladstone) [inline php/class-fm-bylines-cli.php]: nope, this hook is just here so you can do something with the meta data before you do the migration. So if you want to clean it or check for values etc.

File: php/class-fm-bylines-cli.php (hunk 3 above)
Comment (willgladstone) [inline php/class-fm-bylines-cli.php]: Yeah, I'll throw a empty check on the byline.

File: php/class-fm-bylines-cli.php (hunk 4 above)
Comment (willgladstone) [inline php/class-fm-bylines-cli.php]: Cap is only authors. If for some reason they have changed this, this accounts for it here.

File: php/class-fm-bylines-cli.php (hunk 5 above)
Comment (willgladstone) [inline php/class-fm-bylines-cli.php]: Yeah makes more sense here.

File: php/class-fm-bylines-cli.php (hunk 6 above)
Comment (willgladstone) [inline php/class-fm-bylines-cli.php]: yup

**Pull request merged**